Framework/GenAIFramework/Code/Source/UI/AIAssistantWidget.cpp

```diff
@@ -30,21 +34,63 @@ namespace GenAIFramework
         connect(m_newChatWidget, &NewChatWidget::chatCreated, this, &AIAssistantWidget::OnChatCreated);
 
         m_ui->conversations->addTab(m_newChatWidget, "+");
+
+        FeaturesConversationsStore storedChats;
+        UIConversationsBus::BroadcastResult(storedChats, &UIConversationsBus::Events::GetStoredChats);
+
+        for (const auto& chat : storedChats)
+        {
+            const auto& [modelName, providerName, featureName] = chat.second;
+
+            CreateChat(
+                QString::fromStdString(chat.first.c_str()),
+                QString::fromStdString(modelName.c_str()),
+                QString::fromStdString(providerName.c_str()),
+                QString::fromStdString(featureName.c_str()));
+        }
     }
 
     void AIAssistantWidget::OnChatCreated(
         const QString& chatName, const QString& modelName, const QString& providerName, const QString& featureName)
+    {
+        UIConversationsBus::Broadcast(
+            &UIConversationsBus::Events::OnNewChatWidgetCreated,
+            chatName.toStdString().c_str(),
+            modelName.toStdString().c_str(),
+            providerName.toStdString().c_str(),
+            featureName.toStdString().c_str());
+
+        CreateChat(chatName, modelName, providerName, featureName);
+    }
+
+    void AIAssistantWidget::CreateChat(
```

Comment:
Why the split? The method was not that long (and you did not split in the same way for the removal).

Reply:
This is done to not send multiple bus broadcasts when creating chats that are stored. Removal is universal, because only the user can remove a conversation. Adding is not because the system or the user can add the conversation.